fix(Unity): Updated `native` initialization (#12912)
getsentry/sentry-docs: docs/platforms/unity/native-support/index.mdx

<Alert>

> The Unity SDK captures C# exceptions and does not break any builds targeting Nintendo Switch.

Reviewer: Do we need this alert?
It's odd to say "And it doesn't break builds"

How about we say:
The Sentry SDK supports C# and Native crashes on Android, iOS, Windows, macOS and Linux.
For Game Consoles, the SDK supports C# errors. While Sentry supports native crashes on consoles, support for that through the Unity SDK is a work in progress. Reach out via GitHub for the platform you're interested in: [PlayStation](https://github.com/getsentry/sentry-unity/issues/2050), [Switch](https://github.com/getsentry/sentry-unity/issues/2052) and [Xbox](https://github.com/getsentry/sentry-unity/issues/2051)

</Alert>

## Mobile

> The native crash support for Android and iOS is achieved by adding platform-specific SDKs to the generated Xcode and Gradle projects at build time.

Reviewer 2: ```suggestion
The native crash support for Android and iOS can be enabled by adding platform-specific SDKs to the generated Xcode and Gradle projects at build time.
```
